core-services/pdc/src/main/java/org/silverpeas/core/pdc/pdc/service/PdcSearchProcessor.java

```diff
@@ -0,0 +1,139 @@
+package org.silverpeas.core.pdc.pdc.service;
+
+import org.silverpeas.core.admin.user.model.UserDetail;
+import org.silverpeas.core.contribution.contentcontainer.content.ContentInterface;
+import org.silverpeas.core.contribution.contentcontainer.content.ContentManager;
+import org.silverpeas.core.contribution.contentcontainer.content.ContentPeas;
+import org.silverpeas.core.contribution.contentcontainer.content.GlobalSilverContent;
+import org.silverpeas.core.contribution.contentcontainer.content.IGlobalSilverContentProcessor;
+import org.silverpeas.core.contribution.contentcontainer.content.SilverContentInterface;
+import org.silverpeas.core.index.search.SearchQueryProcessor;
+import org.silverpeas.core.index.search.model.QueryDescription;
+import org.silverpeas.core.index.search.model.SearchResult;
+import org.silverpeas.core.index.search.qualifiers.TaxonomySearch;
+import org.silverpeas.core.pdc.pdc.model.AxisValueCriterion;
+import org.silverpeas.core.pdc.pdc.model.SearchContext;
+import org.silverpeas.core.util.ServiceProvider;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import static org.silverpeas.core.contribution.contentcontainer.content
+    .IGlobalSilverContentProcessor.PROCESSOR_NAME_SUFFIX;
+
+/**
+ * @author Nicolas Eysseric
+ */
+@Singleton
+@TaxonomySearch
+public class PdcSearchProcessor implements SearchQueryProcessor {
+
+  @Inject
+  private PdcManager pdcManager;
+
+  @Override
+  public List<SearchResult> process(final QueryDescription query,
+      final List<SearchResult> results) {
+
+    SearchContext pdcContext = new SearchContext(query.getSearchingUser());
+
+    List<AxisValueCriterion> axisValueCriteria =
+        AxisValueCriterion.fromFlattenedAxisValues(query.getTaxonomyPosition());
+    axisValueCriteria.forEach(anAxisValueCriterion->pdcContext.addCriteria(anAxisValueCriterion));
+
+    if (!pdcContext.isEmpty()) {
+      // We get silvercontentids according to the search context, author, components and dates
+      try {
+        List<Integer> contentIds = pdcManager
+            .findSilverContentIdByPosition(pdcContext, new ArrayList<>(query.getWhereToSearch()),
+                null, query.getRequestedCreatedAfter(), query.getRequestedCreatedBefore());
+
+        List<GlobalSilverContent> contents = getGlobalSilverContents(contentIds, pdcContext.getUserId());
+        Collections.sort(contents, cDateDesc);
+
+        return toSearchResults(contents);
+      } catch (Exception e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1166)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either remove or fill this block of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00108)